lab_4_summarization_textrank/main.py

```diff
@@ -120,23 +147,33 @@ def __init__(self) -> None:
         """
         Constructs all the necessary attributes
         """
-        pass
+        super().__init__()
+        self.num = 0
 
     def _learn_indices(self, tokens: tuple[str, ...]) -> None:
         """
         Fills attributes mapping words and integer equivalents to each other
         :param tokens: a sequence of string tokens
         :return:
         """
-        pass
+        for count, token in enumerate(tokens, start=1000):
+            if token not in (self._word2id.keys() and self._id2word.keys()):
+                self._word2id[token] = count
+                self._id2word[count] = token
 
     def encode_sentences(self, sentences: tuple[Sentence, ...]) -> None:
         """
         Enriches the instances of sentences with their encoded versions
         :param sentences: a sequence of sentences
         :return: a list of sentences with their preprocessed versions
         """
-        pass
+        #я запуталась здесь
```

Comment:
You want to firstly check the validity of the arguments, then access its preprocessed version, then compose its encoded version using the corresponding attribute, and this should be it

Reply:
done